Profile sapling key table modifications and paging

isabeltomb opened the pull request:
- Adds paging to the key table by using the material-ui TablePagination React component
- Adds labels to buttons that show when hovering the cursor
- Fixes the action options to stay in the action column. Previously, if the browser window was smaller than the table width, requiring horizontal scrolling, the expanded action options didn't stay in the action column
- Fixes the table to allow the action options for the last key in the table to expand beyond the bottom border of the table
- Adjusts the table navigation padding to align with column names in the key table

**Testing:**
1. Run the splinter-ui using either the biome docker-compose or set the necessary environment variables and use the oauth docker-compose
`docker-compose -f docker-compose-biome.yaml up --build`
OR
`docker-compose -f docker-compose-oauth.yaml up --build`

2. In a browser navigate to `http://localhost:3030`

3. Register/log in and select the profile tab in the left nav bar

4. Add at least 11 keys to the table to test paging

5. Hover cursor over buttons to see labels

Commit:
Add labels to profile sapling buttons

Add title attributes to the buttons in the profile sapling so
information labels appear when hovering.

Labels added to the table navigation buttons, the action buttons
and the new key button.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

Commit:
Fix key table action options

Modify action button in the key table so that the expanding action
options move with the action column.

Also modify css to allow the expanded actions options of the last key
in the table to go outside of the table boundary.

Previously, if the browser window was smaller than the table width,
requiring horizontal scrolling, when hovering the cursor over the action
button the expanded action options were off screen.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

Commit:
Fix table navigation padding

Fix the table navigation padding to align with column names in the
key table.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

Commit:
Fix plus icon on 'new key' button

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

Commit:
Add paging to profile sapling key table

Use the material-ui TablePagination React component to add paging to the
profile sapling key table. The TablePagination component is added to
the KeyTableNav component. Three attributes, rowsPerPage, page, and
onChangePage are added to the the KeyTableNav component to be passed
into the TablePagination component.

The rowsPerPage attribute is added to the KeyTable component to allow
for the page size to be set when using this component.

Two classes, MuiTablePagination-caption and button.MuiButtonBase-root
are added to the css for KeyTableNav to override the default styling
for the TablePagination component.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

shannynalayna reviewed (commented):
Could we set it so that the active key is displayed first in the table? I noticed when updating any key it puts it to the end of the list (which may be a different page). I would expect the activated key to be displayed first (unless being sorted by something) which these don't seem to be

Commit:
Sort user keys to make active key appear first

Sort the user keys after activating and editing keys to make the active
key always appear at the top of the table.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

isabeltomb commented:
@shannynalayna added a commit to this PR to sort the keys in the table to always keep the active key in the first row of the table

peterschwarz commented on saplings/profile/src/Profile.js:
```diff
@@ -83,7 +83,20 @@ export function Profile() {
 
+  const sortKeysActive = (allKeys, publicKey) => {
+    const activeKey = allKeys.find(key => key.public_key === publicKey);
+    const keyIndex = allKeys.findIndex(key => key.public_key === publicKey);
```
Looks like you could just do this search, and then directly index for the activeKey:
```javascript
const keyIndex // same as this line
const activeKey = allKeys[keyIndex];
```

isabeltomb replied:
Fixed
